src/main/java/keycontacts/logic/parser/SortCommandParser.java

```diff
@@ -0,0 +1,67 @@
+package keycontacts.logic.parser;
+
+import static keycontacts.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static keycontacts.logic.parser.CliSyntax.PREFIX_ADDRESS;
+import static keycontacts.logic.parser.CliSyntax.PREFIX_GRADE_LEVEL;
+import static keycontacts.logic.parser.CliSyntax.PREFIX_NAME;
+import static keycontacts.logic.parser.CliSyntax.PREFIX_PHONE;
+
+import keycontacts.logic.commands.SortCommand;
+import keycontacts.logic.parser.exceptions.ParseException;
+import keycontacts.model.student.StudentComparator;
+import keycontacts.model.student.StudentComparator.SortOrder;
+
+/**
+ * Parses input arguments and creates a new SortCommand object.
+ */
+public class SortCommandParser implements Parser<SortCommand> {
+
+    @Override
+    public SortCommand parse(String userInput) throws ParseException {
+        String trimmedInput = userInput.trim();
+        if (trimmedInput.isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, SortCommand.MESSAGE_USAGE));
+        }
+
+        ArgumentList argList = ArgumentTokenizer.tokenizeToList(userInput, PREFIX_NAME, PREFIX_PHONE, PREFIX_ADDRESS,
+                PREFIX_GRADE_LEVEL);
+
+        if (argList.getValue(0).equals("clear")) {
+            return new SortCommand(null);
+        }
```

Comment:
Does this work to clear the sort as intended? Wouldn't this only work if the user typed e.g. `sort n/clear`

Reply:
It does work on my side, you could give it a try on your side.